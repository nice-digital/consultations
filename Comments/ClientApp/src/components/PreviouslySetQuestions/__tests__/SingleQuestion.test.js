import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import { SingleQuestion } from "../SingleQuestion";

test("should fire newQuestion when the Insert button is clicked at consultation level", () => {
	const fakeProps1 = {
		questionText: "fake question text",
		newQuestion: jest.fn(),
		currentDocumentId: "consultation",
		currentConsultationId: 1,
		questionTypeId: 3232,
		questionType: {
			type: "Text",
		},
		allRoles: [
			"Administrator",
			"SCSTeam",
		],
	};
	render(<SingleQuestion {...fakeProps1}/>);
	const button = screen.getByText("Insert");
	fireEvent.click(button);
	expect(fakeProps1.newQuestion).toHaveBeenCalled();
});

test("should fire newQuestion when the Insert button is clicked", () => {
	const fakeProps2 = {
		questionText: "fake question text",
		newQuestion: jest.fn(),
		currentDocumentId: 2,
		currentConsultationId: 1,
		questionTypeId: 3232,
		questionType: {
			type: "Text",
		},
		allRoles: [
			"Administrator",
			"SCSTeam",
		],
	};
	render(<SingleQuestion {...fakeProps2}/>);
	const button = screen.getByText("Insert");
	fireEvent.click(button);
	expect(fakeProps2.newQuestion).toHaveBeenCalled();
});

test("should match the snapshot", () => {
	const fakeProps1 = {
		questionText: "fake question text",
		newQuestion: jest.fn(),
		currentDocumentId: "consultation",
		currentConsultationId: 1,
		questionTypeId: 3232,
		questionType: {
			type: "Text",
		},
		allRoles: [
			"Administrator",
			"SCSTeam",
		],
	};
	const {container} = render(<SingleQuestion {...fakeProps1}/>);
	expect(container).toMatchSnapshot();
});
