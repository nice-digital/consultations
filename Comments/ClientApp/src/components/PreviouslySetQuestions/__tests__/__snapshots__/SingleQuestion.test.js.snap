// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<div>
  <li>
    <div
      class="card"
    >
      <div
        class="grid"
      >
        <div
          data-g="9"
        >
          <div
            class="card__header"
          >
            <p
              class="card__heading"
            >
              fake question text
            </p>
          </div>
          <dl
            class="card__metadata"
          >
            <div
              class="card__metadatum"
            >
              <dt>
                <span
                  class="card__tag tag tag--flush tag--beta"
                >
                  Open Question
                </span>
              </dt>
              <dd
                class="visually-hidden"
              >
                Text
              </dd>
            </div>
            <div
              class="card__metadatum"
            >
              <dt>
                <span
                  class="card__tag tag tag--flush tag--alpha"
                >
                  Administrator
                </span>
              </dt>
              <dd
                class="visually-hidden"
              >
                Administrator
              </dd>
            </div>
            <div
              class="card__metadatum"
            >
              <dt>
                <span
                  class="card__tag tag tag--flush tag--alpha"
                >
                  SCSTeam
                </span>
              </dt>
              <dd
                class="visually-hidden"
              >
                SCSTeam
              </dd>
            </div>
          </dl>
        </div>
        <div
          data-g="3"
        >
          <button
            class="btn btn-small right mr--0"
          >
            Insert
          </button>
        </div>
      </div>
    </div>
  </li>
</div>
`;
